databroker/core.py

```diff
@@ -249,9 +261,17 @@ def get_table(mds, fs, headers, fields=None, fill=True, convert_times=True,
         # cache these attribute look-ups for performance
         start = header['start']
         stop = header['stop']
+        descriptors = header['descriptors']
         if stop is None:
             stop = {}
-        for descriptor in header['descriptors']:
+
+        # shim for back-compat with old data that has no 'primary' descriptor
+        if not [d for d in descriptors if d.get('name') == 'primary']:
```

Comment:
Minor bikesheddy temporary list-removing improvement: 
`if not any(d for d in descriptors if d.get('name') == 'primary'):`